docs/management/commands/update_docs.py

```diff
@@ -175,29 +177,32 @@ def build_doc_release(self, release, force=False):
 
             if self.verbosity >= 2:
                 self.stdout.write(f"  building {builder} ({source_dir} -> {build_dir})")
-            try:
+            sphinx_init = {
+                "srcdir": source_dir,
+                "confdir": source_dir,
+                "outdir": build_dir,
+                "doctreedir": build_dir.joinpath(".doctrees"),
+                "buildername": builder,
                 # Translated docs builds generate a lot of warnings, so send
-                # stderr to stdout to be logged (rather than generating an
-                # email)
-                subprocess.check_call(
-                    [
-                        "sphinx-build",
-                        "-b",
-                        builder,
-                        "-D",
-                        "language=%s" % to_locale(release.lang),
-                        "-j",
-                        "auto",
-                        "-Q" if self.verbosity == 0 else "-q",
-                        str(source_dir),  # Source file directory
-                        str(build_dir),  # Destination directory
-                    ],
-                    stderr=sys.stdout,
-                )
-            except subprocess.CalledProcessError:
+                # stderr to stdout to be logged (rather than generating an email)
+                "warning": sys.stdout,
+            }
+            # Initialize a temporary Sphinx app to load existing config values
+            # from confdir.
+            app = Sphinx(**sphinx_init)
+            extensions = [*app.config.extensions, "docs.builder"]
```

Comment:
Ah, I hadn't realized this was about adding an item to the list of extensions! I've been pouring over the sphinx source code to see if there was a way to extend the config file's extension list rather than override it but there doesn't seem to be anything clean.
I'd suggest using the `Config` class (`from sphinx.config import Config`) to avoid all the extra work that `Sphinx.__init__()` is doing. Something like:

```suggestion
            # Retrieve the extensions from the conf.py so we can append to them
            conf_extensions = Config.read(source_dir).extensions
            extensions = [*conf_extensions, "docs.builder"]
```